### PR TITLE
Fix #10047 - Update strings to correct spelling of 'Add-ons'

### LINKED ISSUE #10047: Change "Addons" to "Add-ons" on mozilla.org
In an effort to be more consistent, we should change "Addons" to "Add-ons" so that we are aligned with AMO.

One example where this will need to be changed is the subnav:

https://www.mozilla.org/en-US/firefox/new/

### PR DESCRIPTION
**Pull request by maxxcrawford**
## Description

Update all instances of "Addons" across the site with "Add-ons" 

## Issue / Bugzilla link

#10047

## Testing

- http://localhost:8000/en-US/firefox/new/
- http://localhost:8000/en-US/firefox/mobile/

_Note - Both instances are in the subnavigation._

### CODE REVIEW

File: bedrock/firefox/templates/firefox/new/desktop/download.html [hunk 1]
```diff
@@ -52,7 +52,7 @@ <h2 class="c-sub-navigation-title">{{ ftl('sub-navigation-firefox-for-desktop',
           <a href="https://support.mozilla.org/products/firefox/{{ referrals }}" data-link-type="nav" data-link-position="subnav" data-link-name="Support">{{ ftl('sub-navigation-support', fallback='navigation-support') }}</a>
         </li>
         <li class="c-sub-navigation-item">
-          <a href="https://addons.mozilla.org/firefox/{{ referrals }}" data-link-type="nav" data-link-position="subnav" data-link-name="Addons">{{ ftl('sub-navigation-addons', fallback='navigation-add-ons') }}</a>
+          <a href="https://addons.mozilla.org/firefox/{{ referrals }}" data-link-type="nav" data-link-position="subnav" data-link-name="Addons">{{ ftl('sub-navigation-add-ons', fallback='sub-navigation-addons') }}</a>
```
Comment (maxxcrawford) [inline bedrock/firefox/templates/firefox/new/desktop/download.html:55]: Note that I'm leaving `data-link-name="Addons"` alone to not break analytics.

File: l10n/en/sub_navigation.ftl [hunk 2]
```diff
@@ -24,12 +24,12 @@ sub-navigation-desktop-beta-and-developer = Desktop { -brand-name-beta } & { -br
 sub-navigation-desktop-nightly = Desktop { -brand-name-nightly }
 sub-navigation-features = Features
 sub-navigation-support = Support
-sub-navigation-addons = Addons
```
Comment (maxxcrawford) [inline l10n/en/sub_navigation.ftl:27]: BLOCKING Question: I was removing this string as its no longer used, but I have it set as the fallback. Do I need to leave it for now?

**Review by maxxcrawford — COMMENTED** [review]
Self-reviewed!

File: l10n/en/sub_navigation.ftl (hunk 2 above)
Comment (craigcook) [reply l10n/en/sub_navigation.ftl:27]: Yeah, I think it needs to still be in the file to work as a fallback, and you can add a comment above it marking it as obsolete like:
`# Obsolete string`

Comment (maxxcrawford) [reply l10n/en/sub_navigation.ftl:27]: @craigcook Revised!

**Review by craigcook — APPROVED** [review]
🩹

**Pull request merged**